### PR TITLE
Refactor and fix disappearing in first render

### PR DESCRIPTION
**Pull request by victorhmp**
#### What is the purpose of this pull request?

Refactor the codebase and fix a bug where the LocaleSwitcher component would disappear in its first render.

#### What problem is this solving?

Use TypeScript and fix the bug mentioned above and reported here: https://app.clubhouse.io/vtex/story/30905/locale-switcher-is-disappearing

#### How should this be manually tested?

https://victormiranda--powerplanet.myvtex.com/?__bindingAddress=www.powerplanetonline.com/es
https://localeswitcher--storecomponents.myvtex.com/

#### Types of changes

* [x] Bugfix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.
* [ ] Technical improvement.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "locale-switcher",
+  "private": true,
```
Comment (Klynger) [inline package.json]: 🤔

File: react/LocaleSwitcher.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+const LocaleSwitcher = () => {
+  const { data, loading, error } = useQuery<LocalesQuery>(Locales)
+  const { culture, emitter } = useRuntime()
+  const [openLocaleSelector, setOpenLocaleSelector] = useState(false)
+
+  const supportedLanguages: string[] =
+    (data &&
+      (data.currentBinding.supportedLocales || data?.languages.supported)) ||
```
Comment (Klynger) [inline react/LocaleSwitcher.tsx]: You're testing if data exists and then you're using the `?` thing (that I don't the name of this feature of js) to get the `.languages.supported`, why not just

```typescript
data?.currentBinding.supportedLocales || data?.languages.supported || []
```
or

```typescript
data && (data.currentBinding.supportedLocales || data.languages.supported)) || []
```
? [just a question mark to finish my question]

File: react/LocaleSwitcher.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    []
+  )
+
+  const getLocale = (locale: string) => {
+    const localeObj = supportedLangs.find(
+      ({ localeId }) => getLabel(localeId) === getLabel(locale)
+    )
+    return (
+      localeObj ??
+      (supportedLangs?.[0] || {
```
Comment (Klynger) [inline react/LocaleSwitcher.tsx]: Didn't know about this, very nice

File: docs/README.md [hunk 4]
```diff
@@ -1,35 +1,43 @@
...
+```json
+"dependencies": {
+  "vtex.locale-switcher": "0.x"
+}
+```
 
-When implementing this component as a block, various inner blocks may be available. The following interface lists the available blocks within `LocaleSwitcher` and describes if they are required or optional.
+2. Add the `locale-switcher` block to your header. For example:
 
 ```json
```
Comment (Klynger) [inline docs/README.md]:
```suggestion
```jsonc
```

File: docs/README.md [hunk 5]
```diff
@@ -1,35 +1,43 @@
...
-  "locale-switcher": {
-    "component": "LocaleSwitcher"
-  }
+"header-row#desktop": {
+  "children": [
+    ...
```
Comment (Klynger) [inline docs/README.md]:
```suggestion
    // ...
```

File: react/LocaleSwitcher.tsx (hunk 2 above)
Comment (victorhmp) [reply react/LocaleSwitcher.tsx]: Nice!

File: package.json (hunk 1 above)
Comment (victorhmp) [reply package.json]: That came from `vtex setup`

File: react/LocaleSwitcher.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  const { data, loading, error } = useQuery<LocalesQuery>(Locales)
+  const { culture, emitter } = useRuntime()
+  const [openLocaleSelector, setOpenLocaleSelector] = useState(false)
+
+  const supportedLanguages: string[] =
+    (data &&
+      (data.currentBinding.supportedLocales || data?.languages.supported)) ||
+    []
+
+  const supportedLangs = supportedLanguages.reduce(
```
Comment (klzns) [inline react/LocaleSwitcher.tsx]: Make this function pure outside of component

File: react/LocaleSwitcher.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+      return acc.concat({
+        text: getLabel(lang),
+        localeId: lang,
+      })
+    },
+    []
+  )
+
+  const getLocale = (locale: string) => {
```
Comment (klzns) [inline react/LocaleSwitcher.tsx]: Same

File: package.json (hunk 1 above)
Comment (victorhmp) [reply package.json]: Removed it just in case :)

File: package.json [hunk 8]
```diff
@@ -1,6 +1,5 @@
 {
   "name": "locale-switcher",
-  "private": true,
```
Comment (klzns) [inline package.json]: You could leave this. This prevents the code to be published to npm

**Pull request merged**